defaults/main.yml

```diff
@@ -567,59 +557,49 @@ reprepro_updates_ubuntu:
     VerifyRelease: '437D05B5'
 
 # Configuration for HTTP server
-reprepro_nginx_server_http:
-  by_role: 'debops.reprepro'
-  enabled: True
-  ssl: False
-  redirect_to_ssl: '{{ reprepro_nginx_access_policy }}'
-  filename: '{{ reprepro_domains[0] + "_http" }}'
-  name: '{{ reprepro_domains }}'
-  root: '{{ reprepro_repository_path }}'
-
-  location:
-    '/': |
-      try_files $uri $uri/ =404;
-      autoindex on;
-      autoindex_exact_size off;
-      autoindex_localtime on;
-
-  location_allow:
-    '/': '{{ reprepro_allow }}'
-
-
-# Configuration for HTTPS server
-reprepro_nginx_server_https:
-  by_role: 'debops.reprepro'
-  enabled: True
-  listen: False
-  filename: '{{ reprepro_domains[0] + "_https" }}'
-  name: '{{ reprepro_domains }}'
-  root: '{{ reprepro_repository_path }}'
-
-  access_policy: '{{ reprepro_nginx_access_policy }}'
-  auth_basic_realm: '{{ reprepro_nginx_auth_realm }}'
-
-  options: |
-    client_max_body_size {{ reprepro_max_body_size }}M;
-
-  location:
-    '/': |
-      try_files $uri $uri/ =404;
-      autoindex on;
-      autoindex_exact_size off;
-      autoindex_localtime on;
-
-    '/upload': |
-      alias {{ reprepro_upload }};
-      dav_access all:rw;
-      dav_methods PUT;
-      limit_except PUT {
-              deny all;
-      }
-
-  location_allow:
-    '/': '{{ reprepro_allow }}'
-    '/upload': '{{ reprepro_allow_upload }}'
+reprepro_nginx_server:
+  - listen 80
+  - server_name {{ reprepro_domain }}
+  - root {{ reprepro_repository_path }}
+  - access_log /var/log/nginx/{{ reprepro_domain }}.log;
+  - error_log /var/log/nginx/{{ reprepro_domain }}.log;
+  - client_max_body_size {{ reprepro_max_body_size }}M;
+  - |
+    location ~ /\. {
+           return 404;
+    }
+  - |
+    location = /favicon.ico {
+           try_files /favicon.ico =204;
+           access_log off;
+           log_not_found off;
+    }
+  - |
+    location / {
+           try_files $uri $uri/ =404;
+           autoindex on;
+           autoindex_exact_size off;
+           autoindex_localtime on;
+           allow 10.0.0.0/16;
```

Comment:
should be do IP filtering here or rely on EC2 security groups? why I ask this is because, we don't do instance level firewalling anywhere, this being an exception might cause confusion